interpreter/text/parser.mly

```diff
@@ -206,12 +206,14 @@ let inline_type_explicit (c : context) x ft at =
 (* Custom annotations *)
 
 let parse_annots (m : module_) : Custom.section list =
+  let open Lexing in
+  let bs = Bytes.to_string (!Annot.cur_buf).lex_buffer in
```

Comment:
I was also unsure. I did some quick test, including some netsted quoted module, and it seemed to work. I will take a second look at it.

Reply:
I read the source code of the Lexing module. I think that as long as flush_input is not called, the buffer is never cleared. Also, we read it at the end of the module parsing, so it will always include the module itself (but maybe not other stuff is this is a wast file. It should not matter).

something like this:

```
let get_source () =
  let open Lexing in
  let lex = !current_lexbuf in
  Bytes.to_string (Bytes.sub lex.lex_buffer lex.lex_start_pos lex.lex_buffer_len)
```

should be correct

Reply:
Actually I was wrong.
I would expect `lex.lex_start_pos` to always be 0 but it isn't. and when the buffer is full, the data before `lex_start_pos` can be overwritten, resulting in an incomplete string.

This issue needs some more thinking

Reply:
this is solved by wrapping the lexbuf